### ISSUE TITLE
Model loss not improving

### ISSUE THREAD

**Issue by ChristianLus** [IPapakis/GCNNMatch#2]
Hello,

I was able to run your pretrained model on a dataset and get decent results but when I attempted to train a new model, I was having trouble with the loss not going down. I am using the train settings attached below:
![trainsettings](https://user-images.githubusercontent.com/43122756/99847449-13832900-2b46-11eb-86a4-8770ce4a3050.JPG)

I kept seeing the loss value bounce from 1, all the way up to 7 repeatedly. I am unsure if there may be an issue with the settings I am using or if I am missing something else. I have the learning rate and weight decay set to 0.001 as was set in the code. I noticed this is different than what is listed in the paper but I don't believe this is the issue. 

Thank you for your time, I really enjoy your work!

**Comment by IPapakis** [comment]
@ChristianLus Thank you. For how many epochs did you run? The pre-trained model was till epoch 11.

**Comment by ChristianLus** [comment]
Hello,

I attempted to train the network for 4 epochs at first but after 2 epochs the loss behavior was slightly worrying so I stopped the training to ask for some guidance. I was getting loss values that would go from a value of 2 to a value of 7 or 8 in the next batch and this trend would continue for a while. I'm not sure if it took a while for your model to start to converge?

I am retraining the model now to see if a different batch size and learning rate will help me out.

**Comment by IPapakis** [comment]
@ChristianLus You can start training for 1 or 2 epochs with "shuffle=False" in "model_training.py". Then stop and resume training with "shuffle=True".

**Comment by ChristianLus** [comment]
@IPapakis Hi I'm not sure what the issue was at the beginning but this current run seems to be training properly. The model did not seem to need the suggestions you applied above for the loss to go down consistently, I'm not sure what I was doing wrong. 

I appreciate your quick response and suggestions! For future attempts, do you think changing the slack variable would help further improve performance?

**Comment by IPapakis** [comment]
@ChristianLus Sure you can try changing the slack variable, I have noticed -0.2 to 0.2 gives good results.
